libs/common/parameters_utils.cpp

```diff
@@ -294,6 +294,38 @@ void _ReadAttributeConnection(
                           outputElement);
 }
 
+// Get an attribute VtValue, accounting for recursive nodeGraph connections
+bool ResolveAttrValue(const UsdAttribute &attr, VtValue &value, const TimeSettings &time)
+{
+    if (!attr)
+        return false;
+
+    if (attr.HasAuthoredConnections()) {
+        // This attribute has connections, we want to resolve NodeGraph connections
+        SdfPathVector targets;
+        attr.GetConnections(&targets);
+        if (!targets.empty()) {
+            SdfPath &target = targets[0]; // just consider the first target
+            SdfPath primPath = target.GetPrimPath();
+            UsdPrim targetPrim = attr.GetPrim().GetStage()->GetPrimAtPath(primPath);
+            if (targetPrim && targetPrim.IsA<UsdShadeNodeGraph>()) {
```

Comment:
I think it should be possible to use UsdShadeUtils, like `GetValueProducingAttributes` :https://openusd.org/release/api/class_usd_shade_utils.html#ad1f99058557167a9282c2531f2332bd6
This recursively goes into nodegraphs and connections and find the attribute with the value.

Reply:
and potentially remove this function